coder-sdk/env.go

```diff
@@ -125,3 +132,52 @@ func (c Client) DialEnvironmentBuildLog(ctx context.Context, envID string) (*web
 func (c Client) DialEnvironmentStats(ctx context.Context, envID string) (*websocket.Conn, error) {
 	return c.dialWs(ctx, "/api/environments/"+envID+"/watch-stats")
 }
+
+// DialResourceLoad opens a websocket connection for cpu load metrics on the environment
+func (c Client) DialResourceLoad(ctx context.Context, envID string) (*websocket.Conn, error) {
+	return c.dialWs(ctx, "/api/environments/"+envID+"/watch-resource-load")
+}
+
+// BuildLogType describes the type of an event.
+type BuildLogType string
+
+const (
+	// BuildLogTypeStart signals that a new build log has begun.
+	BuildLogTypeStart BuildLogType = "start"
+	// BuildLogTypeStage is a stage-level event for an environment.
+	// It can be thought of as a major step in the environment's
+	// lifecycle.
+	BuildLogTypeStage BuildLogType = "stage"
+	// BuildLogTypeError describes an error that has occurred.
+	BuildLogTypeError BuildLogType = "error"
+	// BuildLogTypeSubstage describes a subevent that occurs as
+	// part of a stage. This can be the output from a user's
+	// personalization script, or a long running command.
+	BuildLogTypeSubstage BuildLogType = "substage"
+	// BuildLogTypeDone signals that the build has completed.
+	BuildLogTypeDone BuildLogType = "done"
+)
+
+type buildLogMsg struct {
+	Type BuildLogType `json:"type"`
+}
+
+// WaitForEnvironmentReady will watch the build log and return when done
+func (c Client) WaitForEnvironmentReady(ctx context.Context, env *Environment) error {
```

Comment:
Why does this need the full environment when the above function doesn't?

Reply:
Ah, good point.